include/contractor/contractor_config.hpp

```diff
@@ -39,30 +40,28 @@ namespace osrm
 namespace contractor
 {
 
-struct ContractorConfig
+struct ContractorConfig final : storage::IOConfig
 {
-    ContractorConfig() : requested_num_threads(0) {}
+    ContractorConfig()
+        : requested_num_threads(0), IOConfig(
```

Comment:
there are some compiler warnings here (and at similar places below) because we first initialize the num. of threads and then the parent object - you might want to change the order here